File: src/main/java/com/danyazero/utils/TypeNode.java
package com.danyazero.utils;

import com.danyazero.model.Type;
import com.danyazero.model.ast.Node;

public class TypeNode implements Node {
    private final com.danyazero.model.Type<?> type;

    public TypeNode(com.danyazero.model.Type<?> type) {
        this.type = type;
    }

    @Override
    public void produce(GenerationContext ctx) {

    }

    @Override
    public void resolveTypes(GenerationContext ctx) {

    }

    public Type<?> getType() {
        return type;
    }
}

File: src/main/kotlin/node/TypeNode.kt
package com.danyazero.node

import com.danyazero.model.Node
import com.danyazero.model.Type
import com.danyazero.utils.GenerationContext

class TypeNode(
    val type: Type<*>,
) : Node {
    override fun produce(ctx: GenerationContext) {}

    override fun toString(): String {
        return "TypeNode(type=$type)"
    }

}